node/processingOnlyNode/syncedMessenger.go

```diff
@@ -0,0 +1,354 @@
+		if peer == peerID {
+			return true
+		}
+	}
+
+	return false
+}
+
+// ConnectedPeers returns the same list as the function Peers
+func (messenger *syncedMessenger) ConnectedPeers() []core.PeerID {
+	return messenger.Peers()
+}
+
+// ConnectedAddresses returns all connected addresses
+func (messenger *syncedMessenger) ConnectedAddresses() []string {
+	peers := messenger.network.GetConnectedPeers()
+	addresses := make([]string, 0, len(peers))
+	for _, peer := range peers {
+		addresses = append(addresses, fmt.Sprintf(virtualAddressTemplate, peer.Pretty()))
+	}
+
+	return addresses
+}
+
+// PeerAddresses returns the virtual peer address
+func (messenger *syncedMessenger) PeerAddresses(pid core.PeerID) []string {
+	return []string{fmt.Sprintf(virtualAddressTemplate, pid.Pretty())}
+}
+
+// ConnectedPeersOnTopic returns the connected peers on the provided topic
+func (messenger *syncedMessenger) ConnectedPeersOnTopic(topic string) []core.PeerID {
+	return messenger.network.GetConnectedPeersOnTopic(topic)
+}
+
+// SetPeerShardResolver does nothing and returns nil
+func (messenger *syncedMessenger) SetPeerShardResolver(_ p2p.PeerShardResolver) error {
+	return nil
+}
+
+// GetConnectedPeersInfo return current connected peers info
+func (messenger *syncedMessenger) GetConnectedPeersInfo() *p2p.ConnectedPeersInfo {
+	peersInfo := &p2p.ConnectedPeersInfo{}
+	peers := messenger.network.GetConnectedPeers()
+	for _, peer := range peers {
+		peersInfo.UnknownPeers = append(peersInfo.UnknownPeers, peer.Pretty())
+	}
+
+	return peersInfo
+}
+
+// WaitForConnections does nothing
+func (messenger *syncedMessenger) WaitForConnections(_ time.Duration, _ uint32) {
+}
+
+// IsConnectedToTheNetwork returns true
+func (messenger *syncedMessenger) IsConnectedToTheNetwork() bool {
+	return true
+}
+
+// ThresholdMinConnectedPeers returns 0
+func (messenger *syncedMessenger) ThresholdMinConnectedPeers() int {
+	return 0
+}
+
+// SetThresholdMinConnectedPeers does nothing and returns nil
+func (messenger *syncedMessenger) SetThresholdMinConnectedPeers(_ int) error {
+	return nil
+}
+
+// SetPeerDenialEvaluator does nothing and returns nil
+func (messenger *syncedMessenger) SetPeerDenialEvaluator(_ p2p.PeerDenialEvaluator) error {
+	return nil
+}
+
+// ID returns the peer ID
+func (messenger *syncedMessenger) ID() core.PeerID {
+	return messenger.pid
+}
+
+// Port returns 0
+func (messenger *syncedMessenger) Port() int {
+	return 0
+}
+
+// Sign will return the hash(messenger.ID + payload)
+func (messenger *syncedMessenger) Sign(payload []byte) ([]byte, error) {
+	return hasher.Compute(messenger.pid.Pretty() + string(payload)), nil
+}
+
+// Verify will check if the provided signature === hash(pid + payload)
+func (messenger *syncedMessenger) Verify(payload []byte, pid core.PeerID, signature []byte) error {
+	sig := hasher.Compute(pid.Pretty() + string(payload))
+	if bytes.Equal(sig, signature) {
+		return nil
+	}
+
+	return errors.New("invalid signature")
```

Comment:
might define a variable for this error at the top of the page (`var` section)

Reply:
defined for all errors